thewalrus/fock_gradients.py

```diff
@@ -58,20 +57,43 @@ def displacement(r, phi, cutoff, dtype=np.complex128):  # pragma: no cover
         array[complex]: matrix representing the displacement operation.
     """
     D = np.zeros((cutoff, cutoff), dtype=dtype)
-    sqrt = np.sqrt(np.arange(cutoff, dtype=dtype))
-    mu = np.array([r * np.exp(1j * phi), -r * np.exp(-1j * phi)])
 
-    D[0, 0] = np.exp(-0.5 * r**2)
-    for m in range(1, cutoff):
-        D[m, 0] = mu[0] / sqrt[m] * D[m - 1, 0]
+    factorial = np.arange(0, cutoff)
+    factorial[0] = 1
+    factorial_sqrt = np.cumprod(np.sqrt(factorial))
 
-    for m in range(cutoff):
-        for n in range(1, cutoff):
-            D[m, n] = mu[1] / sqrt[n] * D[m, n - 1] + sqrt[m] / sqrt[n] * D[m - 1, n - 1]
+    for n_minus_m in range(cutoff):
+        m_max = cutoff - n_minus_m
+        L = _laguerre(r**2.0, m_max, n_minus_m)
+
+        alpha_n_minus_m = (r**n_minus_m) * np.exp(1j * phi * n_minus_m)
+        e_r = np.exp(-(r**2.0) / 2.0)
+        for m in range(m_max):
+            n = n_minus_m + m
+            D[n, m] = (factorial_sqrt[m] / factorial_sqrt[n]) * alpha_n_minus_m * e_r * L[m]
```

Comment:
This is like the sqrt of the inverse of the Pocchammer symbol, perhaps `1/np.sqrt(np.cumprod(np.arange(n_minus_m, n+1)))` can be more precise (not sure if more precision helps here)

Reply:
Alternatively we can use a renormalized recurrence relation

Reply:
I'm resolving this in favout of the newer (more numerically stable) apprach using the logs